app/tasks/schedule.php

```diff
@@ -0,0 +1,155 @@
+<?php
+
+global $cli;
+global $register;
+
+use Cron\CronExpression;
+use Utopia\App;
+use Utopia\CLI\Console;
+use Utopia\Database\DateTime;
+use Utopia\Database\Query;
+use Swoole\Timer;
+
+const FUNCTION_VALIDATION_TIMER = 180; //seconds
+const FUNCTION_ENQUEUE_TIMER = 60; //seconds
+const ENQUEUE_TIME_FRAME = 60 * 5; // 5 min
+sleep(4); // Todo prevent PDOException
+
+$cli
+->task('schedule-new')
+->desc('Function scheduler task')
+->action(function () use ($register) {
+    Console::title('Scheduler V1');
+    Console::success(APP_NAME . ' Scheduler v1 has started');
+
+    $createQueue = function () use (&$functions, &$queue) {
+        /**
+         * Creating smaller functions list containing 5-min timeframe.
+         */
+        $timeFrame = DateTime::addSeconds(new \DateTime(), ENQUEUE_TIME_FRAME);
+        foreach ($functions as $function) {
+            $cron = new CronExpression($function['schedule']);
+            $next = DateTime::format($cron->getNextRunDate());
+            if ($next < $timeFrame) {
+                $queue[$next][$function['scheduleId']] = $function;
+            }
+        }
+    };
+
+    $removeFromQueue = function ($scheduleId) use (&$queue) {
+        foreach ($queue as $slot => $schedule) {
+            foreach ($schedule as $function) {
+                if ($scheduleId === $function['scheduleId']) {
+                    unset($queue[$slot][$function['scheduleId']]);
+                }
+            }
+        }
+    };
+
+
+    $dbForConsole = getConsoleDB();
+    $count = 0;
+    $limit = 50;
+    $sum = $limit;
+    $functions = [];
+    $queue = [];
+
+    /**
+     * Initial run fill $functions list
+     */
+    while ($sum === $limit) {
+        $results = $dbForConsole->find('schedules', [
+            Query::equal('region', [App::getEnv('_APP_REGION')]),
+            Query::equal('type', ['function']),
+            Query::equal('active', [true]),
+            Query::limit($limit)
+        ]);
+
+        $sum = count($results);
+        foreach ($results as $document) {
+            $functions[$document['scheduleId']] = $document;
+            $count++;
+        }
+    }
+
+    $createQueue();
+    $lastUpdate =  DateTime::addSeconds(new \DateTime(), -FUNCTION_VALIDATION_TIMER);
+
+    Co\run(
+        function () use ($removeFromQueue, $createQueue, $dbForConsole, &$functions, &$queue, &$lastUpdate) {
+            Timer::tick(FUNCTION_VALIDATION_TIMER * 1000, function () use ($removeFromQueue, $createQueue, $dbForConsole, &$functions, &$queue, &$lastUpdate) {
```

Comment:
If we already use a constant, we can just have it store the result of this calculation.

Reply:
The same constant is used first time for seconds  and the second time for milliseconds